### DISCUSSION TITLE
Can't assign a snapshotconnection to ViewportComponent.

### DISCUSSION THREAD

**Discussion by Snailya** [iTwin/itwinjs-core#4314] [Q&A]
VS Code said that the type of imodel is (string & IModelConnection) | (number & IModelConnection).
![image](https://user-images.githubusercontent.com/19658624/191399643-b60a98b3-0f53-4dd5-b47c-5e46952f43ca.png)
But if I open the ViewportCompoent.d.ts, the type of imodel is IModelConnection.
![image](https://user-images.githubusercontent.com/19658624/191399840-08222dba-f355-4e95-b352-807caeeb39b1.png)
The react itself has a attribute named imodel.
![image](https://user-images.githubusercontent.com/19658624/191400767-368f53b7-1b8d-4f52-af4e-9b9074a2bef4.png)

**Comment by aruniverse** [comment]
Wow, TIL! Mind sharing what version of @types/react you're using?

**Reply by Snailya** [reply]
I'm using the react 18.2.0, @types/react 18.0.20.
I'm not that familiar with dependency version, so I've just installed all the packages without specifying version.
When installing @itwin/appui-react using npm, there's error with dependency resolution. I'm not sure how to solve it, so I delete all the modules and use yarn to reinstall mannully, and at that time seems no error.

**Reply by aruniverse** [reply]
So you're saying switching to yarn from npm fixes the issue for you?

Heads up, AppUi doesn't officially support react@18 right now. It might work, but there also might be bugs we're unaware about. We will be looking to support react@18 in a future release of the appui pkgs.

**Reply by Snailya** [reply]
Switching from npx to yarn only helps to install the appui package. To hack this problem, I commented out the imodel attributes in @types/react
